pkg/rpcserver/local.go

```diff
@@ -97,8 +97,14 @@ func RunLocal(cfg *LocalConfig) error {
 		// If the executor has crashed early, reply to all remaining requests to unblock tests.
 	loop:
 		for {
-			req := serv.execSource.Next()
-			if req == nil {
+			req, stop := serv.execSource.Next()
+			if req != nil {
+				req.Done(&queue.Result{Status: queue.ExecFailure, Err: errors.New("executor crashed")})
+			}
+			if stop {
+				break loop
+			} else if req == nil {
+				// No new requests so far, but there might be.
 				select {
 				case <-cfg.Context.Done():
```

Comment:
We don't need this context if we have stop.

Reply:
I think that shutting `RunLocal()` down must not necessarily depend on the `queue.Source` saying it doesn't have any more inputs -- e.g. the fuzzer is not supposed to even run out of them.

Reply:
I don't follow.

User of RunLocal provides the source. Why would that source return stop when there is no intention to stop? And what's the difference between returning stop the source and cancelling the context? When what should be used? If they mean the same, why do we need 2 ways to express it?